app/src/main/java/cm/aptoide/pt/home/BundlesResponseMapper.java

```diff
@@ -78,8 +85,14 @@ public BundlesResponseMapper(String marketName, InstallManager installManager,
               new Event().setName(Event.Name.getAds), widgetTag));
         } else if (type.equals(HomeBundle.BundleType.INFO_BUNDLE) || type.equals(
             HomeBundle.BundleType.EDITORIAL)) {
-          appBundles.add(new ActionBundle(title, type, event, widgetTag,
-              map((ActionItemResponse) viewObject)));
+          ActionItem actionItem = map((ActionItemResponse) viewObject);
+          appBundles.add(new ActionBundle(title, type, event, widgetTag, actionItem));
+        } else if (type.equals(HomeBundle.BundleType.WALLET_ADS_OFFER)) {
+          ActionItem actionItem = map((ActionItemResponse) viewObject);
+          if (walletAdsOfferCardManager.shouldShowWalletOfferCard(
```

Comment:
I will forever be against these identifier arguments (e.g. shouldShowWalletOfferCard(type+"_"+id). I know we're doing similarly in a lot of other cases so I'm not going to say for you to change it but the reason I don't like it is:
- For us to properly use 'shouldShowWalletOfferCard' and 'blackListManager.addImpression' we need to know the identifier structure, i.e. type + "_" + cardId, which makes using this harder. But if we pass the arguments and let those methods create the structure, we'll lose flexibility.

So I guess what I'm saying is that it's up to you :D